agent/grpc-external/services/peerstream/stream_tracker.go

```diff
@@ -216,38 +203,40 @@ func (s *Status) GetExportedServicesCount() uint64 {
 
 // IsHealthy is a convenience func that returns true/ false for a peering status.
 // We define a peering as unhealthy if its status satisfies one of the following:
-// -  If heartbeat hasn't been received within the IncomingHeartbeatTimeout
-// -  If the last sent error is newer than last sent success
+// -  If it is disconnected
```

Comment:
If we go based on just disconnection then during every peer server upgrade or leadership transition there will be a period where we would report the peering as unhealthy.

This seems a little alaramist to me as these are normal operations. As long as we don't remain disconnected for too long (have received a heartbeat recently enough) we can assume the relationship is healthy.

Reply:
Both options seem reasonable. I'll go with the "received a heartbeat recently enough" for now unless there are other opinions.